Detectors/TOF/reconstruction/src/Decoder.cxx

```diff
@@ -0,0 +1,200 @@
+#include "TOFReconstruction/Decoder.h"
+#include <iostream>
+#include <chrono>
+#include <algorithm>
+#include "CommonConstants/LHCConstants.h"
+#define VERBOSE
+
+namespace o2
+{
+namespace tof
+{
+namespace compressed
+{
+
+bool Decoder::open(std::string name)
+{
+  if (mFile.is_open()) {
+    std::cout << "Warning: a file was already open, closing" << std::endl;
+    mFile.close();
+  }
+  mFile.open(name.c_str(), std::fstream::in | std::fstream::binary);
+  if (!mFile.is_open()) {
+    std::cerr << "Cannot open " << name << std::endl;
+    return true;
+  }
+
+  if (mBuffer)
+    delete[] mBuffer;
+
+  mFile.seekg(0, mFile.end);
+  mSize = mFile.tellg();
+  mFile.seekg(0);
+
+  mBuffer = new char[mSize];
+
+  // read content of infile
+  mFile.read(mBuffer, mSize);
+  mUnion = reinterpret_cast<Union_t*>(mBuffer);
+  mUnionEnd = reinterpret_cast<Union_t*>(mBuffer + mSize - 1);
+  close();
+
+  return false;
+}
+
+bool Decoder::close()
+{
+  if (mFile.is_open())
+    mFile.close();
+  return false;
+}
+
+void Decoder::readTRM(std::vector<Digit>* digits, int iddl, int orbit, int bunchid)
+{
+  if (mVerbose)
+    printTRMInfo();
+  int nhits = mUnion->frameHeader.numberOfHits;
+  int time_ext = mUnion->frameHeader.frameID << 13;
+  int itrm = mUnion->frameHeader.trmID;
+  mUnion++;
+
+  // read hits
+  Int_t channel, echannel;
+  Int_t tdc;
+  Int_t tot;
+  Int_t bc;
+  Int_t time;
+
+  Int_t digitInfo[4];
```

Comment:
could you use std::array<int,4> here which you can pass by reference to FromRawHit2Digit?

Reply:
Yes. Done